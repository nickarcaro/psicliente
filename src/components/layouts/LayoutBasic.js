import React from "react";
import { Link, withRouter } from "react-router-dom";
import { Layout, Menu } from "antd";

function LayoutBasic() {
  const { Header, Content, Footer } = Layout;

  return (
    <Layout>
      <Header style={{ position: "fixed", zIndex: 1, width: "100%" }}>
        <Menu theme="dark" mode="horizontal" defaultSelectedKeys={["/home"]}>
          <Menu.Item key="/home">
            <Link to={"/home"}>Home</Link>
          </Menu.Item>
          <Menu.Item key="/contacto">
            <Link to={"/contacto"}>Contacto</Link>
          </Menu.Item>
        </Menu>
      </Header>
      <Content
        className="site-layout"
        style={{ padding: "0 50px", marginTop: 64 }}
      >
        <div
          className="site-layout-background"
          style={{ padding: 24, minHeight: 380 }}
        ></div>
      </Content>
      <Footer style={{ textAlign: "center" }}>
        Ant Design Â©2018 Created by Ant UED
      </Footer>
    </Layout>
  );
}

export default withRouter(LayoutBasic);
