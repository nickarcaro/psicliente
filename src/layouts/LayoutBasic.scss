.logo {
  width: 120px;
  height: 31px;
  margin: 16px 24px 16px 0;
  float: left;
}
h1 {
  margin-top: 50px;
  text-align: center;
}
.site-layout .site-layout-background {
  background: #fff;
}
[data-theme="dark"] .site-layout .site-layout-background {
  background: #141414;
}

.basicfooter {
  position: fixed;
  bottom: 0;
  width: 100%;
  height: 60px; /* Height of the footer */
}
