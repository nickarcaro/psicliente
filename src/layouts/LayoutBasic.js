import React from "react";
import { Link, withRouter } from "react-router-dom";
import { Layout, Menu } from "antd";
import LoadRoutes from "./LoadRoutes";
import "./LayoutBasic.scss";
import UdpLogo from "../assets/jpeg/logo.jpeg";
function LayoutBasic(props) {
  const { routes } = props;
  const { Header, Content, Footer } = Layout;

  return (
    <Layout className="layout">
      <Header style={{ position: "fixed", zIndex: 1, width: "100%" }}>
        <img className="logo" src={UdpLogo} alt="clinica psicologica udp" />
        <Menu theme="dark" mode="horizontal">
          <Menu.Item key="/home">
            <Link to={"/home"}>home</Link>
          </Menu.Item>
          <Menu.Item key="/contacto">
            <Link to={"/contacto"}>Contacto</Link>
          </Menu.Item>
        </Menu>
      </Header>
      <Content
        className="site-layout"
        style={{ padding: "0 50px", marginTop: 64 }}
      >
        <div className="site-layout-background">
          <h1>Plataforma de gestión de consultantes</h1>
          <LoadRoutes routes={routes} />
        </div>
      </Content>
      <Footer style={{ textAlign: "center" }}>
        Clínica Psicológica UDP ©2020 Created by TICS II
      </Footer>
    </Layout>
  );
}

export default withRouter(LayoutBasic);
