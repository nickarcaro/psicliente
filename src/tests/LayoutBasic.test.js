import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";

import LayoutBasic from "../layouts/LayoutBasic";

describe("pruebas del <layoutbasic/>", () => {
  test("debe mostrar <LayoutBasic/> correctamente", () => {
    const wrapper = shallow(<LayoutBasic />);
    expect(wrapper).toMatchSnapshot();
  });
});
