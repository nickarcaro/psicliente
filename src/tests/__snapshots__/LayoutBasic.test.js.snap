// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas del <layoutbasic/>  debe mostrar <LayoutBasic/> correctamente 1`] = `
<Layout>
  <Header
    style={
      Object {
        "position": "fixed",
        "width": "100%",
        "zIndex": 1,
      }
    }
  >
    <div
      className="logo"
    />
    <Menu
      defaultSelectedKeys={
        Array [
          "2",
        ]
      }
      mode="horizontal"
      theme="dark"
    >
      <MenuItem
        key="1"
      >
        nav 1
      </MenuItem>
      <MenuItem
        key="2"
      >
        nav 2
      </MenuItem>
      <MenuItem
        key="3"
      >
        nav 3
      </MenuItem>
    </Menu>
  </Header>
  <Content
    className="site-layout"
    style={
      Object {
        "marginTop": 64,
        "padding": "0 50px",
      }
    }
  >
    <div
      className="site-layout-background"
      style={
        Object {
          "minHeight": 380,
          "padding": 24,
        }
      }
    >
      <LoadRoutes />
    </div>
  </Content>
  <Footer
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Ant Design Â©2018 Created by Ant UED
  </Footer>
</Layout>
`;

exports[`pruebas del <layoutbasic/> debe mostrar <LayoutBasic/> correctamente 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
